scripts/release.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR="${SCRIPT_DIR}/../"
+cd ${ROOT_DIR}
+
+print_usage() {
+  cat <<EOF
+Usage: $(basename $0) tag"
+
+Tag example: v1.2.3
+EOF
+}
+
+if [[ $# != 1 ]]
+then
+  print_usage
+  exit 1
+fi
+
+release_tag="$1"
+
+import_gpg_keys() {
+  echo ">>> Setting GnuPG configuration ..."
+  mkdir -p ~/.gnupg
+  chmod 700 ~/.gnupg
+  cat > ~/.gnupg/gpg.conf <<EOF
+no-tty
+pinentry-mode loopback
+EOF
+
+  echo ">>> Importing secret key ..."
+  echo "$GPG_SECRET_KEY" > seckey.gpg
+  trap "rm seckey.gpg" EXIT
+  gpg --batch --allow-secret-key-import --import seckey.gpg
+}
+
+build_project() {
+  mkdir -p dist
+
+  echo ">>> Building the javaagent ..."
+  ./gradlew -Prelease.useLastTag=true build final closeAndReleaseSonatypeStagingRepository -x test --no-daemon
```

Comment:
Why no tests? How can we be sure that this given commit is not broken?

Reply:
Test are run in the `build` stage/job - see `.gitlab-ci.yml`. This script is only run once the build passes.

Reply:
Well, except smoke tests - gitlab runners can't into docker containers, so we only run those on GitHub.